src/runtime/entitlements-manager.js

```diff
@@ -244,26 +252,50 @@ export class EntitlementsManager {
       default:
         return;
     }
-    this.postEntitlementsRequest_(new EntitlementJwt(), result, source);
+    const token = parseQueryString(this.win_.location.search)['gaa_n'];
+    let isUserRegistered = null;
+    if (
+      'additionalParameters' in event &&
+      event.additionalParameters !== null &&
+      event.additionalParameters.getIsUserRegistered() !== null
+    ) {
+      isUserRegistered = event.additionalParameters.getIsUserRegistered();
+    }
+    this.postEntitlementsRequest_(
+      new EntitlementJwt(),
+      result,
+      source,
+      token,
+      isUserRegistered
+    );
   }
 
   // Informs the Entitlements server about the entitlement used
   // to unlock the page.
   postEntitlementsRequest_(
     usedEntitlement,
     entitlementResult,
-    entitlementSource
+    entitlementSource,
+    optionalToken = '',
+    optionalIsUserRegistered = null
   ) {
     const message = new EntitlementsRequest();
     message.setUsedEntitlement(usedEntitlement);
     message.setClientEventTime(toTimestamp(Date.now()));
     message.setEntitlementResult(entitlementResult);
     message.setEntitlementSource(entitlementSource);
+    message.setToken(optionalToken);
+    if (optionalIsUserRegistered !== null) {
+      message.setIsUserRegistered(optionalIsUserRegistered);
+    }
 
-    const url =
+    let url =
       '/publication/' +
       encodeURIComponent(this.publicationId_) +
       '/entitlements';
+    if (this.encodedParams_) {
+      url += '?encodedParams=' + this.encodedParams_;
```

Comment:
Could we use the `addQueryParam()` method from `url.js` here?

Reply:
I just sent you a PR that updates the rest of this script to use `addQueryParam()` https://github.com/subscriptions-project/swg-js/pull/1573

Reply:
Thanks for the PR!